lib/internal/checkUrl.js

```diff
@@ -66,6 +85,15 @@ function checkUrl(link, baseUrl, options, callback)
 		}
 		
 		linkObj.clean(link);
+
+		if ( cache !== undefined && cache !== null )
```

Comment:
I am surprised that you're okay with the non-strict equality, but I'm cool with it too. Changed.